test/RachisTests/RavenDB_14348.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading.Tasks;
+using FastTests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace RachisTests
+{
+    public class RavenDB_14348 : RavenTestBase
+    {
+        public RavenDB_14348(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task ShouldWork()
```

Comment:
What are you actually testing here?

Reply:
1. On store dispose, the WaitForIndexNotification doesn't gets timeout exception, because of  '[drain requests code](https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/Documents/DocumentDatabase.cs#L633)'
2. In DeleteDatabaseCommand we assign '[new topology](https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/ServerWide/Commands/DeleteDatabaseCommand.cs#L74)' _without_ databaseId, if we have ClusterTransactions kicking in for that DB, they would throw on '[wrong serialization](https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/ServerWide/ClusterStateMachine.cs#L764)'
3. After db deleted and ClusterTransactions are still kicking in, the db record would be null, and we get NRE.